// @aws-sdk/client-sagemaker-runtime/dist-cjs/schemas/schemas_0.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.InvokeEndpointWithResponseStream$ = exports.InvokeEndpointAsync$ = exports.InvokeEndpoint$ = exports.ResponseStream$ = exports.PayloadPart$ = exports.InvokeEndpointWithResponseStreamOutput$ = exports.InvokeEndpointWithResponseStreamInput$ = exports.InvokeEndpointOutput$ = exports.InvokeEndpointInput$ = exports.InvokeEndpointAsyncOutput$ = exports.InvokeEndpointAsyncInput$ = exports.errorTypeRegistries = exports.ValidationError$ = exports.ServiceUnavailable$ = exports.ModelStreamError$ = exports.ModelNotReadyException$ = exports.ModelError$ = exports.InternalStreamFailure$ = exports.InternalFailure$ = exports.InternalDependencyException$ = exports.SageMakerRuntimeServiceException$ = void 0;
const _A = "Accept";
const _B = "Body";
const _BB = "BodyBlob";
const _By = "Bytes";
const _CA = "CustomAttributes";
const _CAH = "CustomAttributesHeader";
const _CSI = "ClosedSessionId";
const _CT = "ContentType";
const _CT_ = "Content-Type";
const _EC = "ErrorCode";
const _EE = "EnableExplanations";
const _EN = "EndpointName";
const _F = "Filename";
const _FL = "FailureLocation";
const _ICN = "InferenceComponentName";
const _IDE = "InternalDependencyException";
const _IE = "InvokeEndpoint";
const _IEA = "InvokeEndpointAsync";
const _IEAI = "InvokeEndpointAsyncInput";
const _IEAO = "InvokeEndpointAsyncOutput";
const _IEI = "InvokeEndpointInput";
const _IEO = "InvokeEndpointOutput";
const _IEWRS = "InvokeEndpointWithResponseStream";
const _IEWRSI = "InvokeEndpointWithResponseStreamInput";
const _IEWRSO = "InvokeEndpointWithResponseStreamOutput";
const _IF = "InternalFailure";
const _II = "InferenceId";
const _IL = "InputLocation";
const _IPV = "InvokedProductionVariant";
const _ISF = "InternalStreamFailure";
const _ITS = "InvocationTimeoutSeconds";
const _LSA = "LogStreamArn";
const _M = "Message";
const _ME = "ModelError";
const _MNRE = "ModelNotReadyException";
const _MSE = "ModelStreamError";
const _NSI = "NewSessionId";
const _OL = "OutputLocation";
const _OM = "OriginalMessage";
const _OSC = "OriginalStatusCode";
const _PB = "PartBlob";
const _PP = "PayloadPart";
const _RS = "ResponseStream";
const _RTTLS = "RequestTTLSeconds";
const _SI = "SessionId";
const _SOPE = "S3OutputPathExtension";
const _SU = "ServiceUnavailable";
const _TCH = "TargetContainerHostname";
const _TM = "TargetModel";
const _TV = "TargetVariant";
const _VE = "ValidationError";
const _XASA = "X-Amzn-SageMaker-Accept";
const _XASCA = "X-Amzn-SageMaker-Custom-Attributes";
const _XASCSI = "X-Amzn-SageMaker-Closed-Session-Id";
const _XASCT = "X-Amzn-SageMaker-Content-Type";
const _XASEE = "X-Amzn-SageMaker-Enable-Explanations";
const _XASF = "X-Amzn-SageMaker-Filename";
const _XASF_ = "X-Amzn-SageMaker-FailureLocation";
const _XASI = "X-Amzn-SageMaker-InputLocation";
const _XASIC = "X-Amzn-SageMaker-Inference-Component";
const _XASII = "X-Amzn-SageMaker-Inference-Id";
const _XASI_ = "X-Amzn-SageMaker-InvocationTimeoutSeconds";
const _XASNSI = "X-Amzn-SageMaker-New-Session-Id";
const _XASO = "X-Amzn-SageMaker-OutputLocation";
const _XASR = "X-Amzn-SageMaker-RequestTTLSeconds";
const _XASS = "X-Amzn-SageMaker-S3OutputPathExtension";
const _XASSI = "X-Amzn-SageMaker-Session-Id";
const _XASTCH = "X-Amzn-SageMaker-Target-Container-Hostname";
const _XASTM = "X-Amzn-SageMaker-Target-Model";
const _XASTV = "X-Amzn-SageMaker-Target-Variant";
const _aQE = "awsQueryError";
const _c = "client";
const _e = "error";
const _eP = "eventPayload";
const _h = "http";
const _hE = "httpError";
const _hH = "httpHeader";
const _s = "smithy.ts.sdk.synthetic.com.amazonaws.sagemakerruntime";
const _se = "server";
const _st = "streaming";
const _xAIPV = "x-Amzn-Invoked-Production-Variant";
const n0 = "com.amazonaws.sagemakerruntime";
const schema_1 = require("@smithy/core/schema");
const errors_1 = require("../models/errors");
const SageMakerRuntimeServiceException_1 = require("../models/SageMakerRuntimeServiceException");
const _s_registry = schema_1.TypeRegistry.for(_s);
exports.SageMakerRuntimeServiceException$ = [-3, _s, "SageMakerRuntimeServiceException", 0, [], []];
_s_registry.registerError(exports.SageMakerRuntimeServiceException$, SageMakerRuntimeServiceException_1.SageMakerRuntimeServiceException);
const n0_registry = schema_1.TypeRegistry.for(n0);
exports.InternalDependencyException$ = [-3, n0, _IDE,
    { [_e]: _se, [_hE]: 530 },
    [_M],
    [0]
];
n0_registry.registerError(exports.InternalDependencyException$, errors_1.InternalDependencyException);
exports.InternalFailure$ = [-3, n0, _IF,
    { [_e]: _se, [_hE]: 500 },
    [_M],
    [0]
];
n0_registry.registerError(exports.InternalFailure$, errors_1.InternalFailure);
exports.InternalStreamFailure$ = [-3, n0, _ISF,
    { [_e]: _se },
    [_M],
    [0]
];
n0_registry.registerError(exports.InternalStreamFailure$, errors_1.InternalStreamFailure);
exports.ModelError$ = [-3, n0, _ME,
    { [_e]: _c, [_hE]: 424 },
    [_M, _OSC, _OM, _LSA],
    [0, 1, 0, 0]
];
n0_registry.registerError(exports.ModelError$, errors_1.ModelError);
exports.ModelNotReadyException$ = [-3, n0, _MNRE,
    { [_aQE]: [`ModelNotReadyException`, 429], [_e]: _c, [_hE]: 429 },
    [_M],
    [0]
];
n0_registry.registerError(exports.ModelNotReadyException$, errors_1.ModelNotReadyException);
exports.ModelStreamError$ = [-3, n0, _MSE,
    { [_e]: _c },
    [_M, _EC],
    [0, 0]
];
n0_registry.registerError(exports.ModelStreamError$, errors_1.ModelStreamError);
exports.ServiceUnavailable$ = [-3, n0, _SU,
    { [_e]: _se, [_hE]: 503 },
    [_M],
    [0]
];
n0_registry.registerError(exports.ServiceUnavailable$, errors_1.ServiceUnavailable);
exports.ValidationError$ = [-3, n0, _VE,
    { [_e]: _c, [_hE]: 400 },
    [_M],
    [0]
];
n0_registry.registerError(exports.ValidationError$, errors_1.ValidationError);
exports.errorTypeRegistries = [
    _s_registry,
    n0_registry,
];
var BodyBlob = [0, n0, _BB, 8, 21];
var CustomAttributesHeader = [0, n0, _CAH, 8, 0];
var PartBlob = [0, n0, _PB, 8, 21];
exports.InvokeEndpointAsyncInput$ = [3, n0, _IEAI,
    0,
    [_EN, _IL, _CT, _A, _CA, _II, _SOPE, _F, _RTTLS, _ITS],
    [[0, 1], [0, { [_hH]: _XASI }], [0, { [_hH]: _XASCT }], [0, { [_hH]: _XASA }], [() => CustomAttributesHeader, { [_hH]: _XASCA }], [0, { [_hH]: _XASII }], [0, { [_hH]: _XASS }], [0, { [_hH]: _XASF }], [1, { [_hH]: _XASR }], [1, { [_hH]: _XASI_ }]], 2
];
exports.InvokeEndpointAsyncOutput$ = [3, n0, _IEAO,
    0,
    [_II, _OL, _FL],
    [0, [0, { [_hH]: _XASO }], [0, { [_hH]: _XASF_ }]]
];
exports.InvokeEndpointInput$ = [3, n0, _IEI,
    0,
    [_EN, _B, _CT, _A, _CA, _TM, _TV, _TCH, _II, _EE, _ICN, _SI],
    [[0, 1], [() => BodyBlob, 16], [0, { [_hH]: _CT_ }], [0, { [_hH]: _A }], [() => CustomAttributesHeader, { [_hH]: _XASCA }], [0, { [_hH]: _XASTM }], [0, { [_hH]: _XASTV }], [0, { [_hH]: _XASTCH }], [0, { [_hH]: _XASII }], [0, { [_hH]: _XASEE }], [0, { [_hH]: _XASIC }], [0, { [_hH]: _XASSI }]], 2
];
exports.InvokeEndpointOutput$ = [3, n0, _IEO,
    0,
    [_B, _CT, _IPV, _CA, _NSI, _CSI],
    [[() => BodyBlob, 16], [0, { [_hH]: _CT_ }], [0, { [_hH]: _xAIPV }], [() => CustomAttributesHeader, { [_hH]: _XASCA }], [0, { [_hH]: _XASNSI }], [0, { [_hH]: _XASCSI }]], 1
];
exports.InvokeEndpointWithResponseStreamInput$ = [3, n0, _IEWRSI,
    0,
    [_EN, _B, _CT, _A, _CA, _TV, _TCH, _II, _ICN, _SI],
    [[0, 1], [() => BodyBlob, 16], [0, { [_hH]: _CT_ }], [0, { [_hH]: _XASA }], [() => CustomAttributesHeader, { [_hH]: _XASCA }], [0, { [_hH]: _XASTV }], [0, { [_hH]: _XASTCH }], [0, { [_hH]: _XASII }], [0, { [_hH]: _XASIC }], [0, { [_hH]: _XASSI }]], 2
];
exports.InvokeEndpointWithResponseStreamOutput$ = [3, n0, _IEWRSO,
    0,
    [_B, _CT, _IPV, _CA],
    [[() => exports.ResponseStream$, 16], [0, { [_hH]: _XASCT }], [0, { [_hH]: _xAIPV }], [() => CustomAttributesHeader, { [_hH]: _XASCA }]], 1
];
exports.PayloadPart$ = [3, n0, _PP,
    0,
    [_By],
    [[() => PartBlob, { [_eP]: 1 }]]
];
exports.ResponseStream$ = [4, n0, _RS,
    { [_st]: 1 },
    [_PP, _MSE, _ISF],
    [[() => exports.PayloadPart$, 0], [() => exports.ModelStreamError$, 0], [() => exports.InternalStreamFailure$, 0]]
];
exports.InvokeEndpoint$ = [9, n0, _IE,
    { [_h]: ["POST", "/endpoints/{EndpointName}/invocations", 200] }, () => exports.InvokeEndpointInput$, () => exports.InvokeEndpointOutput$
];
exports.InvokeEndpointAsync$ = [9, n0, _IEA,
    { [_h]: ["POST", "/endpoints/{EndpointName}/async-invocations", 202] }, () => exports.InvokeEndpointAsyncInput$, () => exports.InvokeEndpointAsyncOutput$
];
exports.InvokeEndpointWithResponseStream$ = [9, n0, _IEWRS,
    { [_h]: ["POST", "/endpoints/{EndpointName}/invocations-response-stream", 200] }, () => exports.InvokeEndpointWithResponseStreamInput$, () => exports.InvokeEndpointWithResponseStreamOutput$
];
