// @aws-sdk/client-sagemaker-runtime/dist-cjs/models/errors.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ModelStreamError = exports.InternalStreamFailure = exports.ValidationError = exports.ServiceUnavailable = exports.ModelNotReadyException = exports.ModelError = exports.InternalFailure = exports.InternalDependencyException = void 0;
const SageMakerRuntimeServiceException_1 = require("./SageMakerRuntimeServiceException");
class InternalDependencyException extends SageMakerRuntimeServiceException_1.SageMakerRuntimeServiceException {
    name = "InternalDependencyException";
    $fault = "server";
    Message;
    constructor(opts) {
        super({
            name: "InternalDependencyException",
            $fault: "server",
            ...opts,
        });
        Object.setPrototypeOf(this, InternalDependencyException.prototype);
        this.Message = opts.Message;
    }
}
exports.InternalDependencyException = InternalDependencyException;
class InternalFailure extends SageMakerRuntimeServiceException_1.SageMakerRuntimeServiceException {
    name = "InternalFailure";
    $fault = "server";
    Message;
    constructor(opts) {
        super({
            name: "InternalFailure",
            $fault: "server",
            ...opts,
        });
        Object.setPrototypeOf(this, InternalFailure.prototype);
        this.Message = opts.Message;
    }
}
exports.InternalFailure = InternalFailure;
class ModelError extends SageMakerRuntimeServiceException_1.SageMakerRuntimeServiceException {
    name = "ModelError";
    $fault = "client";
    Message;
    OriginalStatusCode;
    OriginalMessage;
    LogStreamArn;
    constructor(opts) {
        super({
            name: "ModelError",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ModelError.prototype);
        this.Message = opts.Message;
        this.OriginalStatusCode = opts.OriginalStatusCode;
        this.OriginalMessage = opts.OriginalMessage;
        this.LogStreamArn = opts.LogStreamArn;
    }
}
exports.ModelError = ModelError;
class ModelNotReadyException extends SageMakerRuntimeServiceException_1.SageMakerRuntimeServiceException {
    name = "ModelNotReadyException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "ModelNotReadyException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ModelNotReadyException.prototype);
        this.Message = opts.Message;
    }
}
exports.ModelNotReadyException = ModelNotReadyException;
class ServiceUnavailable extends SageMakerRuntimeServiceException_1.SageMakerRuntimeServiceException {
    name = "ServiceUnavailable";
    $fault = "server";
    Message;
    constructor(opts) {
        super({
            name: "ServiceUnavailable",
            $fault: "server",
            ...opts,
        });
        Object.setPrototypeOf(this, ServiceUnavailable.prototype);
        this.Message = opts.Message;
    }
}
exports.ServiceUnavailable = ServiceUnavailable;
class ValidationError extends SageMakerRuntimeServiceException_1.SageMakerRuntimeServiceException {
    name = "ValidationError";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "ValidationError",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ValidationError.prototype);
        this.Message = opts.Message;
    }
}
exports.ValidationError = ValidationError;
class InternalStreamFailure extends SageMakerRuntimeServiceException_1.SageMakerRuntimeServiceException {
    name = "InternalStreamFailure";
    $fault = "server";
    Message;
    constructor(opts) {
        super({
            name: "InternalStreamFailure",
            $fault: "server",
            ...opts,
        });
        Object.setPrototypeOf(this, InternalStreamFailure.prototype);
        this.Message = opts.Message;
    }
}
exports.InternalStreamFailure = InternalStreamFailure;
class ModelStreamError extends SageMakerRuntimeServiceException_1.SageMakerRuntimeServiceException {
    name = "ModelStreamError";
    $fault = "client";
    Message;
    ErrorCode;
    constructor(opts) {
        super({
            name: "ModelStreamError",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ModelStreamError.prototype);
        this.Message = opts.Message;
        this.ErrorCode = opts.ErrorCode;
    }
}
exports.ModelStreamError = ModelStreamError;
